explorer/interpreter/resolve_names.cpp

```diff
@@ -576,7 +576,8 @@ static auto ResolveNames(Declaration& declaration, StaticScope& enclosing_scope,
       auto& function = cast<CallableDeclaration>(declaration);
       StaticScope function_scope;
       function_scope.AddParent(&enclosing_scope);
-      enclosing_scope.MarkDeclared(function.name());
+      std::string name(*GetName(declaration));
```

Comment:
Just to be sure this doesn't become an issue, we can add the following

```suggestion
      const auto name = GetName(declaration);
      CARBON_CHECK(name) << "Unexpected missing name for `" << declaration << "`.";
```